### PR TITLE
Merge Tag Product List and Product List

### PR DESCRIPTION
**Pull request by ErdemT09**
image

I have removed the TaggedProductList component. Instead of it, what we now do is:

Try to receive a tag using `useParams()`
Call fetchItems with along this tag.
If tag is defined, that method returns the dispatch action method for the tagged product list. Otherwise, the function for the all products.

At the end, the same products state is edited. So, single source of truth for multiple actions.

### CODE REVIEW

File: src/redux/reducers/product-list-reducer.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 export default (state = { products: [], totalCount:0 }, action) => {
     switch (action.type) {
         case 'UPDATE_PRODUCT_LIST':
-            return { ...state, products: [...state.products, ...action.products], totalCount:action.totalCount };
+            return { ...state, products: action.products, totalCount:action.totalCount };
```
Comment (ErdemT09) [inline src/redux/reducers/product-list-reducer.js:4]: @altayhunoglu What would you think about that change? Should we be adding to the list or replacing the list with the input data? For tags, we have to replace. But there will be other cases, I assume.

File: src/components/products/ProductList.js [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
 import { Row, Col } from 'react-bootstrap';
-import { fetchProducts } from '../../redux/actions/product-list-action.js';
+import { fetchItems } from '../../redux/actions/product-list-action.js';
 import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
 import TagPanel from '../TagPanel.js';
 
 const PAGE_ITEMS_COUNT = 9;
 
 const ProductList = ({ fetchProducts, products, addToCart, totalProductCount }) => {
```
Comment (altayhunoglu) [inline src/components/products/ProductList.js:11]:
```suggestion
const ProductList = ({ fetchItems, products, addToCart, totalProductCount }) => {
```

File: src/components/products/ProductList.js [hunk 3]
```diff
@@ -32,19 +36,19 @@ const ProductList = ({ fetchProducts, products, addToCart, totalProductCount })
   }, [totalProductCount, pageNumber]);
 
   useEffect(() => {
-    fetchProducts(pageNumber);
-  }, [pageNumber]);
+    fetchProducts(pageNumber, tag);
```
Comment (altayhunoglu) [inline src/components/products/ProductList.js:39]:
```suggestion
    fetchItems(pageNumber, tag);
```

**Comment by altayhunoglu** [comment]
Great achievement Erdem, thank you very much.

**Review by altayhunoglu — APPROVED** [review]
Great achievement Erdem, thank you very much.

**Pull request merged**